webapp/src/ts/services/cht-script-api.service.ts

```diff
@@ -27,16 +30,32 @@ export class CHTScriptApiService {
     return this.initialized;
   }
 
-  private init() {
+  private async init() {
     this.watchChanges();
     this.userCtx = this.sessionService.userCtx();
-    return this.getSettings();
+    await this.getSettings();
+    await this.loadScripts();
   }
 
-  private getSettings() {
-    return this.settingsService
-      .get()
-      .then(settings => this.settings = settings);
+  private async loadScripts() {
+    const extensionLibs = this.settings?.extension_libs;
```

Comment:
I feel a little uncomfortable about getting this info from `settings`. It means you need to make changes in two places. This will be obscured by cht-conf make it a single command, but it still feels fragile.

The other option would be to replicate the libs into couch, but I feel better about those being part of the service worker like other static resources, rather than having a loader of their own.

If you can think of a better alternative let me know!